### PR TITLE
PD-314: Update Badge component to new palette

### PR DESCRIPTION
**Pull request by DesignerDave**
## ✍️ Proposed changes

This PR updates the Badge component to reflect recent changes to the designer sketch library.

🎟 _Jira ticket:_ [Update Badge component to new palette](https://jira.mongodb.org/browse/PD-314)

## 🛠 Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/badge/README.md [hunk 1]
```diff
@@ -27,9 +24,9 @@ import Badge from '@leafygreen-ui/badge';
 
 **Type:** `string`
 
-**Default:** `'default'`
+**Default:** `'LightGray'`
 
-Sets the style variant of the badge. Valid variants for badges are `'default'`, `'danger'`, `'warning'`, `'darkBlue'`, `'lightBlue'`, `'primary'`, `'outline'`, and `'dark'`.
+Sets the style variant of the badge. Valid variants for badges are `'LightGray'`, `'DarkGray'`, `'Red'`, `'Blue'`, `'Green'`, and `'Yellow'`
```
Comment (hswolff) [inline packages/badge/README.md]: These should be lowercase to match the actual values.

**Comment by DesignerDave** [comment]
What actual colors are you referring to? They aren’t just a solid color
anymore and all of them are composed of 3 divvy colors in the hue range.

File: packages/badge/README.md (hunk 1 above)
Comment (hswolff) [reply packages/badge/README.md]: These are the Variant's keys. We want the list to be the Variant's values.

File: packages/badge/README.md [hunk 2]
```diff
@@ -7,18 +7,15 @@
 ```Javascript
 import Badge from '@leafygreen-ui/badge';
 
-  <Badge
-    variant='lightBlue'
-    className='my-badge'
-  >
-    New
-  </Badge>
+<Badge variant='Blue' className='my-badge'>
```
Comment (hswolff) [inline packages/badge/README.md]: Ditto here. Either Variant.Blue or 'blue'.

Comment (DesignerDave) [reply packages/badge/README.md]: 🤦‍♂ Whoops – fixed in all instances.

**Pull request merged**